src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,112 @@
 import React from 'react';
+import { Todo } from '../../types/Todo';
+import { Filter } from '../../types/Filter';
 
-export const TodoList: React.FC = () => (
-  <table className="table is-narrow is-fullwidth">
-    <thead>
-      <tr>
-        <th>#</th>
-        <th>
-          <span className="icon">
-            <i className="fas fa-check" />
-          </span>
-        </th>
-        <th>Title</th>
-        <th> </th>
-      </tr>
-    </thead>
+type Props = {
+  todos: Todo[];
+  selectedTodo: number | null;
+  selectTodo: (todoId: number) => void;
+  filter: string;
+  search: string;
+};
 
-    <tbody>
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  selectedTodo: selectedTodoId,
+  selectTodo,
+  filter,
+  search,
+}) => {
+  const todosForRender = todos
+    .filter((todo: Todo) => {
+      switch (filter) {
+        case Filter.all:
+          return todo;
+        case Filter.completed:
+          return todo.completed;
+        case Filter.active:
+          return !todo.completed;
+        default:
+          throw new Error('Something went wrong with the filtering');
+      }
+    })
```

Comment:
//[HINT] if you need to filter your items by selected filter - use switch/case statement in correct way.
// Perform the necessary filtering only on a case-by-case basis.

// GOOD

const filteredCards = useMemo(() => {
  switch (filter) {
    case CardFilters.All: {
      return cards;
    }

    case CardFilters.Active: {
      return cards.filter(({ deletedDate }) => !deletedDate);
    }

    case CardFilters.Deleted: {
      return cards.filter(({ deletedDate }) => Boolean(deletedDate));
    }

    default:
      return [];
  }
}, [cards, filter]);

// Don't filter your array right away with switch/case inside your callback.
// Because in case CardFilters.All your program just iterate an array and doesn't do any useful actions. So it's a wasting of memory and CPU (very bad)

// BAD

const filteredCards = useMemo(() => {
  return cards.filter(({ deletedDate }) => {
    switch (filter) {
      case CardFilters.All: {
        return true;
      }

      case CardFilters.Active: {
        return !deletedDate;
      }

      case CardFilters.Deleted: {
        return Boolean(deletedDate);
      }

      default:
        return false;
    }
  });
}, [cards, filter]);